README.md

```diff
@@ -148,6 +164,24 @@ In the example above, `directoryCid` resolves to an IPFS directory with the foll
 └── src
     └── main.py
 ```
+### Alternate implementation options
+
+As discussed above, there are options outside the traditional client-server model that w3up supports. We how to use `w3up-client` to achieve these options in this section.
+
+#### Direct upload from client
+
+In this option, you (the developer) own your Space, but delegate permissions to your users to directly upload content to the service on your behalf. This isn’t completely “serverless” - you still need some infrastructure to create delegated UCAN tokens, but it’s minimal, and potentially saves a ton of bandwidth and overhead.
+
+- _To-do: Include instructions_
```

Comment:
Something like:
- First follow the instructions in [https://github.com/web3-storage/w3up-client#creating-a-client-object](https://github.com/web3-storage/w3up-client#creating-a-client-object), [https://github.com/web3-storage/w3up-client#creating-and-registering-spaces](https://github.com/web3-storage/w3up-client#creating-and-registering-spaces)
- Net new step: Delegate
    - Can have your user’s `w3up-client` instance create a Space keypair using the client’s `createSpace` ([https://github.com/web3-storage/w3up-client#createspace](https://github.com/web3-storage/w3up-client#createspace))
        - You can persist this Space keypair in the client’s state (e.g., store the keypair in your backend user database and send it back to your user when they login) or generate a new Space for each session
    - Your user should make a request to you when they want to upload something, and you can create a UCAN delegating upload permission to that Space DID: [https://github.com/web3-storage/w3up-client#createdelegation](https://github.com/web3-storage/w3up-client#createdelegation)
    - From there, they can use that UCAN token to upload content directly to the service by using [https://github.com/web3-storage/w3up-client#addproof](https://github.com/web3-storage/w3up-client#addproof) and one of the upload methods (*****************************************once the proof is added, does the client know to use it for the upload methods?*****************************************)
    - If you want to know what upload each user made, you should have them make a request to you in parallel with the upload’s CID and other info you want to know about their upload
- Then follow the instructions to upload here: [https://github.com/web3-storage/w3up-client#uploading-data](https://github.com/web3-storage/w3up-client#uploading-data)

Reply:
just saying coming soon for now, saving these notes for later